core/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java

```diff
@@ -149,6 +150,17 @@ protected ScriptMetaData createTestInstance() {
     }
 
     @Override
+    protected MutateFunction<ScriptMetaData> getMutateFunction() {
+        return instance -> {
+            // ScriptMetaData doesn't allow us to see the scripts inside it so
+            // the best we can do here is create a new random instance and rely
+            // on the fact that the new instance is very unlikely to be equal to
+            // the old one
+            return createTestInstance();
```

Comment:
I thought about that but I came to the conclusion that adding that negates the purpose of the test, in that this mutate function is used to create an instance that is not equal to the original and then checks that `original.equals(mutated) == false`. By adding the `assumeFalse` line here it means that if there is an error in equals that means two objects are equals when they shouldn't be, we will not see a failure as the test will be skipped due to that very same fact.

I ran the test 10,000 times and found a bug that if it selects 0 scripts in both the original and the mutated version they will be equal and this is relatively likely, so I have pushed a commit that addresses this by ensuring the mutated version always has at least 1 scripts. I ran the modified version 100,000 times without failure